collect_app/src/main/java/org/odk/collect/android/projects/AutomaticProjectCreatorDialog.kt

```diff
@@ -143,15 +139,11 @@ class AutomaticProjectCreatorDialog : MaterialFullScreenDialogFragment() {
         val projectCreatedSuccessfully = projectCreator.createNewProject(CompressionUtils.decompress(result.text))
 
         if (projectCreatedSuccessfully) {
+            ActivityUtils.startActivityAndCloseAllOthers(activity, MainMenuActivity::class.java)
             ToastUtils.showLongToast(getString(R.string.new_project_created))
-            listener?.onProjectAdded()
```

Comment:
It's hard to tell but the logic here is actually the same. The call on `onProjectAdded` made the `SplashScreenActivity` launch the `MainMenuActivity` and close all others.

This approach has the same downside -- the first launch screen is briefly shown before the main menu activity.